server/src/remote.rs

```diff
@@ -82,126 +113,189 @@ impl FlightService for FlightServiceHandler {
-        let fd = FlightDescriptor::new_path(table_names);
-        let fi = FlightInfo::new(IpcMessage(vec![]), Some(fd), vec![], -1, -1);
-        let output = futures::stream::once(async { Ok(fi) });
+        let flight_descriptor = FlightDescriptor::new_path(table_names);
+        let flight_info =
+            FlightInfo::new(IpcMessage(vec![]), Some(flight_descriptor), vec![], -1, -1);
+        let output = stream::once(async { Ok(flight_info) });
         Ok(Response::new(Box::pin(output)))
     }
 
+    /// Not implemented.
     async fn get_flight_info(
         &self,
         _request: Request<FlightDescriptor>,
     ) -> Result<Response<FlightInfo>, Status> {
-        Err(Status::unimplemented("Not yet implemented"))
+        Err(Status::unimplemented("Not implemented."))
     }
 
+    /// Provide the schema of a table in the catalog. The name of the table must
+    /// be provided as the first element in `FlightDescriptor.path`.
     async fn get_schema(
         &self,
         request: Request<FlightDescriptor>,
     ) -> Result<Response<SchemaResult>, Status> {
-        if let Some(table_name) = request.into_inner().path.get(0) {
-            let session = self.context.session.clone();
-            if let Some(catalog) = session.catalog("datafusion") {
-                // default catalog.
-                if let Some(schema) = catalog.schema("public") {
-                    // default schema.
-                    if let Some(table) = schema.table(table_name) {
-                        let schema = &*table.schema();
-                        let options = IpcWriteOptions::default();
-                        let schema_as_ipc = SchemaAsIpc::new(schema, &options);
-                        if let Ok(sr) = schema_as_ipc.try_into() {
-                            Ok(Response::new(sr))
-                        } else {
-                            Err(Status::internal("unable to serialize schema"))
-                        }
-                    } else {
-                        Err(Status::not_found("table does not exist"))
-                    }
-                } else {
-                    Err(Status::internal("schema does not exist"))
-                }
-            } else {
-                Err(Status::internal("catalog does not exist"))
-            }
-        } else {
-            Err(Status::invalid_argument("no table was provided"))
-        }
+        let flight_descriptor = request.into_inner();
+        let table_name = flight_descriptor
+            .path
+            .get(0)
+            .ok_or_else(|| Status::invalid_argument("No table name was provided."))?;
+        let schema = self.get_table_schema_from_default_catalog(table_name)?;
+
+        let options = IpcWriteOptions::default();
+        let schema_as_ipc = SchemaAsIpc::new(&*schema, &options);
+        let serialized_schema = schema_as_ipc
+            .try_into()
+            .map_err(|error| Status::internal(format!("{}", error)))?;
```

Comment:
To convert the error to a `String` so it can be passed to [`Status::internal`](https://docs.rs/tonic/latest/tonic/struct.Status.html#method.internal). I have replaced `format!({},...` with `.to_string()` to make the code more clear. It was also necessary to add a few additional type annotations after switching to `to_string()`.